classes/observers/listeners/BatchMetadataChangedListener.inc.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file classes/observers/listeners/BatchMetadataChangedListener.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class BatchMetadataChangedListener
+ * @ingroup core
+ *
+ * @brief Listener fired when submission metadata's changed on batch
+ */
+
+namespace PKP\observers\listeners;
+
+use PKP\Jobs\Metadata\BatchMetadataChangedJob;
+use PKP\observers\events\BatchMetadataChanged;
+
+class BatchMetadataChangedListener
```

Comment:
I'd be happy to discuss naming patterns for event and listener classes for consistency. I recommend avoiding the last part `Listener` in the class name as the namespace already includes that information. The same for events,  `PublishedEvent` could be renamed to something like `PublicationPublished` (that was created by me before taking namespaces into consideration). I also believe that listener should perform only one task and its action should be reflected in the class name, so one event could have many listeners but we should avoid creating a universal `multitasked`  listener; but I don't have a strong opinion here. What do you think?

Reply:
We could create a forum to discuss this kind of naming and come with some standard rules. Because, as Uncle Bob mentioned on 'Clean Code', naming things are hard.

I usually opt to use `Listener` as part of the class name to avoid problems during autocomplete and `use` dependency injections.

The `PublishedEvent` was named thus to become more generic across OMP/OJS/OPS.

And I agree with the rule of "One Handle Action per Listener".